src/engine/UI/DisplayLayout.cpp

```diff
@@ -7,8 +7,8 @@ using namespace admirals::UI;
 DisplayLayout::DisplayLayout()
     : m_font(Texture::loadFromPath("assets/font.png")) {}
 
-void DisplayLayout::AddElement(std::shared_ptr<Element> element) {
-    m_elements.push_back(std::move(element));
+void DisplayLayout::AddElement(const std::shared_ptr<Element> &element) {
+    this->m_collection.insert(element);
```

Comment:
The move here was meant to "take" ownership of the context, which is more efficient. So it should be `std::move(element)` inside the insert.